### PR TITLE
docs(*): add sip for moving wit files outside Spin repo

### PR DESCRIPTION
**Pull request by michelleN**
Signed-off-by: Michelle Dhanani <michelle@fermyon.com>

### CODE REVIEW

File: docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# title = "SIP 012 - Maintaining WIT files in a separate repository" template = "main" date = "2023-01-25T01:01:01Z"
+
+Summary: This SIP outlines how to maintain WIT files that Spin (and Spin SDKs consume) outside of the Spin repository.
```
Comment (radu-matei) [inline docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md:3]: nit: I assume "consume" should be outside the paranthesis?

File: docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Owners: michelle@fermyon.com
+
+Created: January 25, 2023
+
+## Background
+
+The WIT files in the `wit/` directory of the current Spin repository define the interfaces that are imported and exported by the Spin SDKs. More specifically, they are referenced by path by the `wit-bindgen` tooling to generate language specific bindings for the SDKs. Example below.
```
Comment (radu-matei) [inline docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md:11]: nit: link to `wit-bindgen`?

File: docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+2. The pull request is reviewed.
+3. The pull request is merged.
+4. After being merged, a github action is used to automate updating the `fermyon/spin-wit` repository using `git subtree push`.
+
+The pull request being merged into the Spin repository may or may not contain additional changes outside of the wit directory. This shouldn't matter as the `git subtree push` command will filter out all irrelevant changes. The `fermyon/spin-wit` repository never needs to be modified by anything other than the github action and all consumer repositories can pull in latest updates with the `git subtree pull` command.
+
+## Future design considerations
+
+- Consider flattening the structure of the `wit/` directory by gettind rid of the `wit/emphemeral` directory to more closely conform to the WIT package structure as described by the component model [here](https://github.com/WebAssembly/component-model/pull/141/files#diff-4853dcfce4501ba0f387ca3885f38ac65dc38cc79e4ef16192213d94bce28517R11) and more specifically described [here](https://github.com/WebAssembly/component-model/pull/141/files#diff-4853dcfce4501ba0f387ca3885f38ac65dc38cc79e4ef16192213d94bce28517R188):
```
Comment (radu-matei) [inline docs/content/sips/012-maintaining-wit-files-in-a-separate-repository.md:55]: I would replace `ephemeral` with `v1` perhaps?
Or something that could evolve once we start talking about a potential `v2` for given interfaces.

**Comment by itowlson** [comment]
Sorry, this is on me for the comment about git subtrees but the SIP seems quite narrowly focused on moving the WIT files, whereas the issue is more broadly about moving the SDKs.  It would be good to understand how things like versioning and examples and tests and templates and stuff are going to work with SDKs in their own repos (e.g. currently we tag SDKs as part of the Spin release workflow).  Managing the WIT files is an important part of that and it's great to have this writeup, but it would be super valuable to nail down the broader proposal.  Sorry and thanks!

**Comment by michelleN** [comment]
@itowlson - no worries. I went back and forth on whether I should include the language SDK move as part of this SIP or as a separate SIP and then wrote this as a separate SIP since it can be an independent change. My plan was to have a follow up SIP for the SDK move. Would you prefer me to merge the SIPs here or do a follow up one for the SDK move?

**Comment by itowlson** [comment]
Ah cool, no worries either way then.  Because the WIT move is motivated by the SIP move I'd be inclined to consider them together, but two separate SIPs could keep things straighter as well!  Thanks for clarifying!

**Comment by lann** [comment]
I'm not super keen on moving these out of `spin`; it would be another thing that could get out of sync with Spin itself. One alternative would be to make the `spin` repo a submodule in SDK repos and have a language-specific "generate" script to produce the bindings for that SDK, like we do for Go currently.

**Comment by michelleN** [comment]
Closing as this is not relevant right now.

**Pull request closed without merging**